shm/Map.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
+
+class Map {
+private:
+    Island* currentPosition_ {};
+    std::vector<Island> islandsLocations_;
+public:
+    void setCurrentPosition(Island* const currentPosition);
+    void addIsland(const Island& newIsland);
```

Comment:
Actually these functions were not required in Task4, but it's probably good to have them.

Reply:
Yea I got ahead of myself. I'm sure they will be needed at some point in this or slightly changed form